Sources/SwiftGenKit/Parsers/Strings/StringsParser.swift

```diff
@@ -38,35 +54,89 @@ public enum Strings {
 
   public final class Parser: SwiftGenKit.Parser {
     private let options: ParserOptionValues
+    private var parsers = [String: StringsFileTypeParser.Type]()
     var tables = [String: [Entry]]()
     public var warningHandler: Parser.MessageHandler?
 
+    private static let subParsers: [StringsFileTypeParser.Type] = [
+      StringsFileParser.self,
+      StringsDictFileParser.self
+    ]
+
     public init(options: [String: Any] = [:], warningHandler: Parser.MessageHandler? = nil) throws {
       self.options = try ParserOptionValues(options: options, available: Parser.allOptions)
       self.warningHandler = warningHandler
+
+      for parser in Parser.subParsers {
+        register(parser: parser)
+      }
     }
 
-    public static let defaultFilter = "[^/]\\.strings$"
     public static let allOptions: ParserOptionList = [Option.separator]
+    public static var defaultFilter: String {
+      let extensions = Parser.subParsers
+        .flatMap { subParser in
+          subParser.extensions.map { NSRegularExpression.escapedPattern(for: $0) }
+        }
+        .sorted()
+        .joined(separator: "|")
+      return "[^/]\\.(?i:\(extensions))$"
+    }
 
     // Localizable.strings files are generally UTF16, not UTF8!
     public func parse(path: Path, relativeTo parent: Path) throws {
+      guard let parserType = parsers[path.extension?.lowercased() ?? ""] else {
+        throw ParserError.unsupportedFileType(path: path, supported: parsers.keys.sorted())
+      }
+
       let name = path.lastComponentWithoutExtension
+      let parser = parserType.init(options: options)
 
-      guard tables[name] == nil else {
-        throw ParserError.duplicateTable(name: name)
-      }
-      guard let data = try? path.read() else {
-        throw ParserError.failureOnLoading(path: path.string)
-      }
+      let entries: [Entry]
+      if let existingEntries = tables[name], !existingEntries.isEmpty {
+        let newEntries = try parser.parseFile(at: path)
+        let allEntries = [existingEntries, newEntries].joined()
+        let allKeys = allEntries.map { $0.key }
+
+        var duplicateKeys = [String]()
+        let mergedEntries = Dictionary(zip(allKeys, allEntries)) { existing, new in
+          duplicateKeys.append(existing.key)
+          return parser.shouldOverwriteValuesInExistingTable ? new : existing
+        }
 
-      let plist = try PropertyListSerialization.propertyList(from: data, format: nil)
-      guard let dict = plist as? [String: String] else {
-        throw ParserError.invalidFormat
+        if !duplicateKeys.isEmpty && !parser.shouldOverwriteValuesInExistingTable {
+          warningHandler?(
+            ParserError.duplicateKeysInTable(
```

Comment:
Doesn't that mean that if the user lists `[Localizable.stringsdict, Localizable.strings]` as the files in its config file, especially the `stringsdict` being listed first, then we will parse the `.stringsdict` first, creating the first entries for `tables[name]`, and then when we come to parsing the `.strings` file, the entries won't be overridden (because of line 104 telling to use `existing` value)… but we'll have a warning anyway?

Given that it's still common (even if unused at runtime) to have the same key in both – especially for people who used SwiftGen <6.2 up until now and used the workaround of declaring plural keys _also_ in their `.strings` files to work around the fact that SwiftGen wasn't able to parse `.stringsdict` files until your PR – I'm not sure that this warning is actually useful after all, might introduce more noise than it's worth?


Note that this warning could also happen even if the user didn't explicitly list the path to the `.stringsdict` files then the path to the `.strings` file in their config file: I think it could also happen if they just point to a `.lproj` folder in their config as I'm not sure the order in which we parse files that are inside folders is deterministic or not.

Reply:
Yes in that case it will log the warning. I can remove it again, I just thought it's better being explicit about potentially changed behavior and it would also point out that the user can clean up their normal `.strings` file. But I'm also fine with adding a paragraph to some migration documentation

Reply:
We do have a Migration Guide that we indeed plan to update when we'll release the next version; I think it makes more sense to put mention of that here indeed 👍